src/main/java/com/ipcounter/classes/CounterStorage.java

```diff
@@ -0,0 +1,58 @@
+package com.ipcounter.classes;
+
+import java.util.BitSet;
+
+/**
+ * Class that can store any number of unique valid IPs
+ */
+public final class CounterStorage {
+
+  /*
+   * BitSet is the best type I found for storing the flags (visited or not) of IPs
+   * It requires integers as indexes so I need two of them becase the number of
+   * unique valid IPs equals exactly 2 * Integer.MAX_VALUE
+   */
+  private BitSet beforeZero;
+  private BitSet afterAndIncludingZero;
+
+  public CounterStorage() {
+    beforeZero = new BitSet(Integer.MAX_VALUE);
+    afterAndIncludingZero = new BitSet(Integer.MAX_VALUE);
+  }
+
+  /**
+   * Marks IP as existing
```

Comment:
marks AND convert string to long. It is not according to SOLID. Storage should store and nothing else.

Reply:
My priority here was providing a good method that didn't expose the inner workings. Class users mustn't care _how_ IPs are stored, and here the storage method requires conversion, which is done by a different class anyways.